src/app/tests/DevToys.UnitTests/Tools/Converters/DateConverterGuiToolTests.cs

```diff
@@ -0,0 +1,497 @@
+        _timeSecondsInputNumber.Text.Should().Be(expectedDateTime.Second.ToString());
+        _timeMillisecondsInputNumber.Text.Should().Be("0");
+    }
+
+    [Theory(DisplayName = "Convert valid timestamp with unix epoch and Milliseconds format should return valid dateTime")]
+    [InlineData(0, "UTC", "1970-01-01T00:00:00.0000000Z")]
+    [InlineData(1700683087000, "UTC", "2023-11-22T19:58:07.0000000Z")]
+    [InlineData(-2180836913000, "UTC", "1900-11-22T19:58:07.0000000Z")]
+    [InlineData(0, "Pacific Standard Time", "1969-12-31T16:00:00.0000000Z")]
+    [InlineData(1700683087000, "Pacific Standard Time", "2023-11-22T11:58:07.0000000Z")]
+    [InlineData(-2180836913000, "Pacific Standard Time", "1900-11-22T11:58:07.0000000Z")]
+    public async Task ConvertValidTimestampWithUnixEpochAndMillisecondsFormatShouldReturnValidDateTime(
+        long timestamp,
+        string timeZoneString,
+        string exceptedDateTimeString)
+    {
+        var expectedDateTime = DateTimeOffset.Parse(exceptedDateTimeString);
+
+        int timezoneIndex = Array.FindIndex(_timeZoneDropDownList.Items, w => w.Value.Equals(timeZoneString));
+        _timeZoneDropDownList.Select(timezoneIndex);
+
+        var formatDropdownList = _formatSetting.InteractiveElement as IUISelectDropDownList;
+        int formatIndex = Array.FindIndex(formatDropdownList.Items, w => w.Value.Equals(DateFormat.Milliseconds));
+        formatDropdownList.Select(formatIndex);
+
+        _timeStampInputText.Text(timestamp.ToString());
+
+        await _tool.WorkTask;
+
+        _timeYearInputNumber.Text.Should().Be(expectedDateTime.Year.ToString());
+        _timeMonthInputNumber.Text.Should().Be(expectedDateTime.Month.ToString());
+        _timeDayInputNumber.Text.Should().Be(expectedDateTime.Day.ToString());
+        _timeHourInputNumber.Text.Should().Be(expectedDateTime.Hour.ToString());
+        _timeMinuteInputNumber.Text.Should().Be(expectedDateTime.Minute.ToString());
+        _timeSecondsInputNumber.Text.Should().Be(expectedDateTime.Second.ToString());
+        _timeMillisecondsInputNumber.Text.Should().Be("0");
+    }
+
+    #endregion
+
+    [Theory(DisplayName = "Convert valid dateTime should return valid Ticks")]
+    [InlineData("1870-01-01T00:00:00.0000000+00:00", "UTC", 589799232000000000)]
+    [InlineData("1923-11-23T19:58:07.0000000+00:00", "UTC", 606806062870000000)]
+    [InlineData("1800-11-22T19:58:07.0000000+00:00", "UTC", 567990862870000000)]
+    [InlineData("1870-01-01T00:00:00.0000000-08:00", "Pacific Standard Time", 589799520000000582)]
+    [InlineData("1870-01-01T00:00:00.0000000+00:00", "Pacific Standard Time", 589799232000000000)]
+    [InlineData("1923-11-23T19:58:07.0000000Z", "Pacific Standard Time", 606806062870000000)]
+    [InlineData("1800-11-22T19:58:07.0000000Z", "Pacific Standard Time", 567990574870000000)]
+    public async Task ConvertValidDateTimeShouldReturnValidTicks(
```

Comment:
Some of these tests fail on my machine. I know you mentioned in private some race condition you're fighting, but I found this one failing even when I run it individually:

![image](https://github.com/veler/DevToys/assets/3747805/e2148459-b21a-4e50-b1b5-dd7ce1d4b9c6)

Not sure what's going on. Let me know if you'd like me to debug it on my own.

Reply:
Yeah should be fixed in the next push

Reply:
It works, thank you! :D